src/test/ui/manually_drop_attr/manually_drop-nodestructor.rs

```diff
@@ -0,0 +1,59 @@
+//! A test of `#[manually_drop]` on a type that *doesn't* have a `Drop` impl.
+//!
+//! The mirror image of `manually_drop-destructor.rs`
+
+#![feature(manually_drop_attr)]
+// run-pass
+extern crate core;
+use core::cell::Cell;
+
+struct DropCounter<'a>(&'a Cell<isize>);
+impl<'a> Drop for DropCounter<'a> {
+    fn drop(&mut self) {
+        self.0.set(self.0.get() + 1);
+    }
+}
+
+#[manually_drop]
+struct ManuallyDropped<'a> {
+    field_1: DropCounter<'a>,
+    field_2: DropCounter<'a>,
+}
+
+#[manually_drop]
+enum ManuallyDroppedEnum<'a> {
+    A(DropCounter<'a>, DropCounter<'a>),
```

Comment:
can you put a variant in front of the variant used in the tests? otherwise the test may pass even though we incorrectly treat enums as if they were structs.